src/Microsoft.VisualStudio.Composition.Analyzers/Utils.cs

```diff
@@ -1,26 +1,278 @@
+    }
+
+    /// <summary>
+    /// Determines whether the specified attribute type matches the expected type and namespace, including inheritance hierarchy.
+    /// </summary>
+    /// <param name="attributeType">The attribute type to check.</param>
+    /// <param name="attributeTypeName">The expected attribute type name.</param>
+    /// <param name="expectedNamespace">The expected namespace components.</param>
+    /// <returns><see langword="true"/> if the attribute type matches; otherwise, <see langword="false"/>.</returns>
+    private static bool IsAttributeOfType(INamedTypeSymbol attributeType, string attributeTypeName, ReadOnlySpan<string> expectedNamespace)
+    {
+        // Check the attribute type itself and its base types
+        INamedTypeSymbol? current = attributeType;
+        while (current is not null)
+        {
+            if (current.Name == attributeTypeName && IsNamespaceMatch(current.ContainingNamespace, expectedNamespace))
+            {
+                return true;
+            }
+
+            current = current.BaseType;
+        }
+
+        return false;
     }
+
+    /// <summary>
+    /// Determines whether the specified symbol has any MEF export attributes. Members of the symbol (if any) are <em>not</em> checked.
+    /// </summary>
+    /// <param name="symbol">The symbol to check for export attributes.</param>
+    /// <returns><see langword="true"/> if the symbol has export attributes; otherwise, <see langword="false"/>.</returns>
+    internal static bool HasExportAttribute(ISymbol symbol)
+    {
+        foreach (AttributeData attribute in symbol.GetAttributes())
+        {
+            if (IsExportAttribute(attribute.AttributeClass))
+            {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Determines whether the specified type has any instance-level MEF exports.
+    /// </summary>
+    /// <param name="symbol">The type symbol to analyze.</param>
+    /// <returns><see langword="true"/> if the type has instance exports; otherwise, <see langword="false"/>.</returns>
+    /// <remarks>
+    /// This method checks for export attributes on the type itself or on its instance members (properties, methods, fields).
+    /// Static members with export attributes are ignored since they don't require type instantiation.
+    /// </remarks>
+    internal static bool HasInstanceExports(INamedTypeSymbol symbol)
+    {
+        // Check the type itself for Export attributes
+        if (Utils.HasExportAttribute(symbol))
+        {
+            return true;
+        }
+
+        // Check instance members (properties, methods, fields) for Export attributes
+        foreach (ISymbol member in symbol.GetMembers())
+        {
+            // Skip static members and nested types
+            if (member.IsStatic || member is ITypeSymbol)
+            {
+                continue;
+            }
+
+            if (Utils.HasExportAttribute(member))
```

Comment:
Will this pick up on `InheritedExportAttribute` inherited from base members? I'm not sure whether an interface implementing method would inherit attributes from the interface member, or if this only applies to `override` members.